src/modules/accommodations/accommodations.service.ts

```diff
@@ -63,6 +73,99 @@ export class AccommodationsService {
     };
   }
 
+  async createAccommodation(
+    createDto: CreateAccommodationDto,
+    userId: string,
+  ): Promise<Accommodation> {
+    const { amenities, categoryId, ...accommodationData } = createDto;
+    const allAmenities = await this.amenitiesService.getAllAmenities();
+    const resolvedAmenities = amenities
+      ? allAmenities.filter((amenity) => amenities.includes(amenity.id))
+      : [];
+
+    const allCategories = await this.categoryService.getAllCategories();
+    const resolvedCategory = allCategories.find(
+      (category) => category.id === categoryId,
+    );
+
+    if (!resolvedCategory) {
+      throw new NotFoundException(`Category with ID ${categoryId} not found`);
+    }
+
+    const newAccommodation = this.accommodationRepository.create({
+      ...accommodationData,
+      amenities: resolvedAmenities,
+      category: resolvedCategory,
+      user: { id: userId },
+    });
+
+    return this.accommodationRepository.save(newAccommodation);
+  }
+
+  async getAccommodationById(id: string): Promise<Accommodation> {
+    const accommodation = await this.accommodationRepository.findOne({
+      where: { id },
+      relations: ['amenities', 'category', 'owner'],
+    });
+
+    if (!accommodation) {
+      throw new NotFoundException(`Accommodation with ID ${id} not found`);
+    }
+
+    return accommodation;
+  }
+
+  async updateAccommodation(
+    id: string,
+    updateDto: UpdateAccommodationDto,
+    userId: string,
+  ): Promise<Accommodation> {
+    const { amenities, categoryId, ...updateData } = updateDto;
+
+    const accommodation = await this.getAccommodationById(id);
+    if (accommodation.user.id !== userId) {
+      throw new UnauthorizedException(
+        'You are not authorized to update this accommodation',
+      );
+    }
+    if (amenities) {
+      const allAmenities = await this.amenitiesService.getAllAmenities();
+      accommodation.amenities = allAmenities.filter((amenity) =>
+        amenities.includes(amenity.id),
+      );
+    }
+    if (categoryId) {
+      const allCategories = await this.categoryService.getAllCategories();
+      const category = allCategories.find((cat) => cat.id === categoryId);
+
+      if (!category) {
+        throw new NotFoundException(`Category with ID ${categoryId} not found`);
+      }
+
+      accommodation.category = category;
+    }
+
+    Object.assign(accommodation, updateData);
+
+    return this.accommodationRepository.save(accommodation);
+  }
+
+  async deleteAccommodation(id: string, userId: string): Promise<void> {
+    const accommodation = await this.accommodationRepository.findOne({
+      where: { id },
+    });
+
+    if (!accommodation) {
+      throw new NotFoundException(`Accommodation with id ${id} not found`);
+    }
+
+    if (accommodation.user.id !== userId) {
```

Comment:
Admin should also have permission to delete.

Reply:
we dont have admin

Reply:
We do.
![image](https://github.com/user-attachments/assets/7015d655-0562-4b31-b5d3-b542a86fcff1)
![image](https://github.com/user-attachments/assets/1049cc64-85e7-4500-84a1-37991e280180)